### ISSUE TITLE
Error typed.scss

### ISSUE THREAD

**Issue by Profi-web** [brandonmcconnell/scss-typewriter#13]
Not compatible with the latest version of Sass, which is disallowing naming a function "--typed-percent-calc." It also mentions an error in the CSS code that is generated by the library, which is not a valid selector.

**Comment by brandonmcconnell** [comment]
@Profi-web Thanks for the report. Could you please share screenshots for each of the errors you're seeing and provide steps on how to reproduce them?

**Comment by brandonmcconnell** [comment]
@nex3 Do you have any suggestions here? My understanding was that starting a function/variable name with `__` (e.g. `__typed-percent-calc`) was the official way to denote private members in Sass ([docs](https://sass-lang.com/documentation/at-rules/use#private-members)).

Has that now changed? If not, any idea why this error would be occurring? If so, is there a backward-compatible solution to this?

**Comment by brandonmcconnell** [comment]
@Profi-web Just following up. I am revisiting that issue now, and I am still unsure of how to reproduce the error. Can you help with that?

@nex3 any idea would could be causing this?

**Comment by findcongwang** [comment]
I'm also seeing something similar. Error message
> Invalid CSS after "...ed-percent-calc": expected selector, was "(.1, 2, .08, 1, 6.0"

**Comment by brandonmcconnell** [comment]
@findcongwang Hey! I've been looking for someone who can help me to reproduce these errors, as I haven't been able to pinpoint them myself yet.

Would you be available to discuss them? For starters, could you give me the code you used which resulted in your error?

**Comment by sumitjaisingh** [comment]
I'm not sure why no one provided you with the code they are trying. Here is what i did

h2 {
        font-size: clamp(16px, 3vw, 36px);
        font-weight: 500;
        font-family: var(--sj-body-font);
        @include typed("Hello World");
    }

also here is my package.json entries as they might help you figure out the issue

    "browser-sync": "^2.26.13",
    "cssnano": "^4.1.10",
    "eslint": "^8.45.0",
    "eslint-plugin-react": "^7.32.2",
    "gulp": "^4.0.2",
    "gulp-postcss": "^9.0.0",
    "gulp-sass": "^4.1.0",
    "gulp-terser": "^1.4.0",
    "postcss": "^8.1.2"

![image](https://github.com/brandonmcconnell/typed.css/assets/14229019/f5834ce1-870d-4c5a-99f1-fcf5af648566)

hope this helps.

**Comment by nex3** [comment]
@brandonmcconnell Sorry I missed this. The issue here is twofold:

1. For historical reasons, Sass considers `_` and `-` synonymous in Sass identifiers.
2. Because `calc()` used to only be available under vendor prefixes, Sass considers anything of the form `-...-calc()` to be calculation syntax with a vendor prefix.

The net result of this is that you can't write a function that begins with `_` and ends with `-calc()`. The easiest solution is probably just to rename it `calculate()` instead.

**Comment by brandonmcconnell** [comment]
@sumitjaisingh @nex3 Thank you both so much for the added context and pointers. I'll get a PR in for this shortly, implementing @nex3's suggested fix, renaming `--typed-percent-calc` to `--typed-percent-calculate` 🙂

**Comment by niklasbogensperger** [comment]
EDIT: I wrote this yesterday but hadn't sent it or refreshed the page, so I didn't see people already gave feedback. Sorry about that.

**TL;DR:** renaming `@function __typed-percent-calc` to `@function __typed-percent-cal` (or to something else not equal to "calc") fixes the issue. It doesn't immediately have something to do with the underscores, but in my testing leaving those out also fixes the issue.

I came across this issue while trying to incorporate this project into my personal website I am currently trying to build with Hugo.
It's been a while since I've done anything (very basic) in HTML and I've never used SCSS before so I was stumped.
I solved it in my case in the end (after several other suggested bugfixes at other locations in the code) by a simple rename of the function in question, as there is a deprecation warning about this exact naming. The rename works by either not having the word "calc" at the end, or leaving it the same and removing the underscores (although arguably the former method would be preferable I think). I have been unable to find the documentation for this change, however.

I can give some context and a minimal example in case you're still searching for some feedback:

I am using Hugo (hugo v0.115.3+extended darwin/amd64) with the "Osprey Delight" theme (https://github.com/kdevo/osprey-delight), which allows directly dropping in custom SCSS files, on a MacOS system. My minimal code (in case it is at all relevant) and fix/output is as follows:

### Code

./assets/sass/_custom.scss:
```scss
@import 'typed';

.cursorheading {
    @include typed("Test string", "Foo", "Bar", null, (caret: true));
}
```

./layouts/partials/body-top.html (partial html for the template):
```html
<header ...>
  <div ...>
     <h1 class="cursorheading"></h1>
  </div>
</header>
```

### Before fix:

typed.scss (row 11):
```scss
@function __typed-percent-calc(...) {
```

Output (Hugo SCSS-to-CSS compiler):
```
DEPRECATION WARNING on line 11 of ./assets/sass/typed.scss:
Naming a function "--typed-percent-calc" is disallowed and will be an error in future versions of Sass.
This name conflicts with an existing CSS function with special parse rules.


<a lot more warnings and verbose output, which made the above hard to spot>


ERROR Rebuild failed: TOCSS: failed to transform "sass/main.scss" (text/x-scss): "./assets/sass/typed.scss:1:1": Invalid CSS after "...ed-percent-calc": expected selector, was "(.1, 2, .08, 1, 6.0"
```

### After fix:

typed.scss (row 11):
```scss
@function __typed-percent-cal(...) {
```

All works as expected, and the other errors that the builtin linter in VS Code was throwing at other lines in the code are also gone.

**Comment by brandonmcconnell** [comment]
@niklasbogensperger Want to open a PR for your fix? 🙂

**Comment by brandonmcconnell** [comment]
Resolved by #18

**Comment by brandonmcconnell** [comment]
Thanks for submitting the PR, @niklasbogensperger!
